### PR TITLE
Fix in_tail input messages loss when receive_lines fail. fix #699

### LINKED ISSUE #699: Will in_tail input loss messages when router emitting fail
In in_tail.rb, the final emitting codes are as follows.

``` ruby
  227     def receive_lines(lines, tail_watcher)
  228       es = @receive_handler.call(lines, tail_watcher)
  229       unless es.empty?
  230         tag = if @tag_prefix || @tag_suffix
  231                 @tag_prefix + tail_watcher.tag + @tag_suffix
  232               else
  233                 @tag
  234               end
  235         begin
  236           router.emit_stream(tag, es)
  237         rescue
  238           # ignore errors. Engine shows logs and backtraces.
  239         end
  240       end
  241     end
```

I wonder if router emitting fail will cause messages to be lost.

### PR DESCRIPTION
**Pull request by repeatedly**
Based on #700. Update patch and rebase to master.

### CODE REVIEW

**Comment by repeatedly** [comment]
@sonots @tagomoris @zheolong How about this?

**Comment by zheolong** [comment]
@repeatedly I think it's great.

File: lib/fluent/plugin/in_tail.rb [hunk 1]
```diff
@@ -258,10 +258,15 @@ def receive_lines(lines, tail_watcher)
...
+        rescue BufferQueueLimitError
+          return false
         rescue
-          # ignore errors. Engine shows logs and backtraces.
+          # ignore non BufferQueueLimitError errors because in_tail can't recover. Engine shows logs and backtraces.
+          return true
         end
       end
+
+      return true
```
Comment (sonots) [inline lib/fluent/plugin/in_tail.rb]: true/false is ambiguous what it means. please write a yardoc comment for the return value.

**Comment by sonots** [comment]
commented

**Comment by repeatedly** [comment]
If there is no problem, I will merge this patch today and release new fluentd.

**Pull request merged**